nebula_decoders/src/nebula_decoders_continental/decoders/continental_ars548_decoder.cpp

```diff
@@ -0,0 +1,574 @@
+
+  radar_status_.lateral_acceleration_status =
+    sensor_status_packet.lateral_acceleration_status == 0   ? "VDY_OK"
+    : sensor_status_packet.lateral_acceleration_status == 1 ? "VDY_NOTOK"
+                                                            : "INVALID VALUE";
+
+  radar_status_.yaw_rate_status = sensor_status_packet.yaw_rate_status == 0   ? "VDY_OK"
+                                  : sensor_status_packet.yaw_rate_status == 1 ? "VDY_NOTOK"
+                                                                              : "INVALID VALUE";
+
+  radar_status_.steering_angle_status = sensor_status_packet.steering_angle_status == 0 ? "VDY_OK"
+                                        : sensor_status_packet.steering_angle_status == 1
+                                          ? "VDY_NOTOK"
+                                          : "INVALID VALUE";
+
+  radar_status_.driving_direction_status =
+    sensor_status_packet.driving_direction_status == 0   ? "VDY_OK"
+    : sensor_status_packet.driving_direction_status == 1 ? "VDY_NOTOK"
+                                                         : "INVALID VALUE";
+
+  radar_status_.characteristic_speed_status =
+    sensor_status_packet.characteristic_speed_status == 0   ? "VDY_OK"
+    : sensor_status_packet.characteristic_speed_status == 1 ? "VDY_NOTOK"
+                                                            : "INVALID VALUE";
+
+  if (sensor_status_packet.radar_status == 0) {
+    radar_status_.radar_status = "STATE_INIT";
+  } else if (sensor_status_packet.radar_status == 1) {
+    radar_status_.radar_status = "STATE_OK";
+  } else if (sensor_status_packet.radar_status == 2) {
+    radar_status_.radar_status = "STATE_INVALID";
+  } else {
+    radar_status_.radar_status = "INVALID VALUE";
+  }
+
+  if (sensor_status_packet.voltage_status == 0) {
+    radar_status_.voltage_status = "Ok";
+  }
+  if (sensor_status_packet.voltage_status & 0x01) {
+    radar_status_.voltage_status += "Current under voltage";
+  }
+  if (sensor_status_packet.voltage_status & 0x02) {
+    radar_status_.voltage_status = "Past under voltage";
+  }
+  if (sensor_status_packet.voltage_status & 0x03) {
+    radar_status_.voltage_status = "Current over voltage";
+  }
+  if (sensor_status_packet.voltage_status & 0x04) {
+    radar_status_.voltage_status = "Past over voltage";
+  }
+
+  if (sensor_status_packet.temperature_status == 0) {
+    radar_status_.temperature_status = "Ok";
+  }
+  if (sensor_status_packet.temperature_status & 0x01) {
+    radar_status_.temperature_status += "Current under temperature";
+  }
+  if (sensor_status_packet.temperature_status & 0x02) {
+    radar_status_.temperature_status += "Past under temperature";
+  }
+  if (sensor_status_packet.temperature_status & 0x03) {
+    radar_status_.temperature_status += "Current over temperature";
+  }
+  if (sensor_status_packet.temperature_status & 0x04) {
+    radar_status_.temperature_status += "Past over temperature";
+  }
```

Comment:
* The values `0x03, 0x04` are incorrect and should be `0x04, 0x08` respectively (bitfield, see radar IO datasheet). 
* For voltage, there can also be any combination of the four error messages and they should thus be concatenated.
* Overvoltage etc. should be spelled as one word
* When concatenating, separating the entries with commas would be nice. This is a pain in CPP but there is a Boost function: `boost::algorithm::join(..)`. Alternatively, trailing commas would also be acceptable imo